onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -188,6 +195,52 @@ namespace py = pybind11;
 using namespace onnxruntime;
 using namespace onnxruntime::logging;
 
+// Custom op section starts
+static Env& platform_env = Env::Default();
+
+CustomOpLibrary::CustomOpLibrary(const char* library_path, OrtSessionOptions& ort_so) {
+  {
+    OrtPybindThrowIfError(platform_env.LoadDynamicLibrary(library_path, &library_handle_));
+
+    if (!library_handle_)
+      throw std::runtime_error("RegisterCustomOpsLibrary: Failed to load library");
+
+    OrtStatus*(ORT_API_CALL * RegisterCustomOps)(OrtSessionOptions * options, const OrtApiBase* api);
+
+    OrtPybindThrowIfError(platform_env.GetSymbolFromLibrary(library_handle_, "RegisterCustomOps", (void**)&RegisterCustomOps));
+
+    if (!RegisterCustomOps)
+      throw std::runtime_error("RegisterCustomOpsLibrary: Entry point RegisterCustomOps not found in library");
+
+    auto* status = RegisterCustomOps(&ort_so, OrtGetApiBase());
+
+    if (status) {
+      // A non-nullptr indicates some error
+      // Free status and throw
+      platform_env.UnloadDynamicLibrary(library_handle_);
+      ::free(status);
+      throw std::runtime_error("TODO");
+    }
+
+    // No status to free if it is a nullptr
+  }
+}
+
+// Unload the library when the destructor is triggered
+CustomOpLibrary::~CustomOpLibrary() {
+  platform_env.UnloadDynamicLibrary(library_handle_);
```

Comment:
unload the library when the corresponding struct that holds it goes out of scope